pkg/helm/build.go

```diff
@@ -29,8 +33,86 @@ const getKubectl string = `RUN apt-get update && \
  mv kubectl /usr/local/bin && \
  chmod a+x /usr/local/bin/kubectl`
 
+// BuildInput represents stdin passed to the mixin for the build command.
+type BuildInput struct {
+	Config MixinConfig
+}
+
+// MixinConfig represents configuration that can be set on the helm mixin in porter.yaml
+// mixins:
+// - helm:
+//	  repositories:
+//	    stable:
+//		  url: "https://kubernetes-charts.storage.googleapis.com"
+//		  cafile: "path/to/cafile"
+//		  certfile: "path/to/certfile"
+//		  keyfile: "path/to/keyfile"
+//		  username: "username"
+//		  password: "password"
+type MixinConfig struct {
+	Repositories map[string]Repository
+}
+
+type Repository struct {
+	URL      string `yaml:"url,omitempty"`
+	Cafile   string `yaml:"cafile,omitempty"`
+	Certfile string `yaml:"certfile,omitempty"`
+	Keyfile  string `yaml:"keyfile,omitempty"`
+	Username string `yaml:"username,omitempty"`
+	Password string `yaml:"password,omitempty"`
+}
+
 func (m *Mixin) Build() error {
+
+	// Create new Builder.
+	var input BuildInput
+	err := builder.LoadAction(m.Context, "", func(contents []byte) (interface{}, error) {
+		err := yaml.Unmarshal(contents, &input)
+		return &input, err
+	})
+	if err != nil {
+		return err
+	}
+
+	// Define helm
 	fmt.Fprintf(m.Out, getHelm, helmDownloadURL)
+
+	// Define kubectl
 	fmt.Fprintf(m.Out, getKubectl, kubeVersion)
+
+	// Go through repositories
+	for name, repo := range input.Config.Repositories {
+
+		commandValue, err := GetAddRepositoryCommand(name, repo.URL, repo.Cafile, repo.Certfile, repo.Keyfile, repo.Username, repo.Password)
+		if err != nil && m.Debug {
+			fmt.Fprintf(m.Out, err.Error())
```

Comment:
I believe Porter has a convention of logging debug statements to the `Err` stream, so this can be updated to `fmt.Fprintf(m.Err, err.Error())`